src/Framework/BuildCheck/IBuildCheckResult.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Experimental.BuildCheck;
+
+/// <summary>
+/// Holder for the reported result of a build cop rule.
+/// </summary>
+public interface IBuildCheckResult
+{
+    /// <summary>
+    /// Optional location of the finding (in near future we might need to support multiple locations).
```

Comment:
If the data being encapsulated may be extended in the future, using an interface tends to be a bad choice. What would we lose if this interface didn't exist?

Reply:
Marking this `internal` to prevent the future breaking. 
It is otherwise needed as it's used by BuildEventArgs infra in Build.Framework assembly - and the actuall implementation need to have some additional info (e.g. locations of type ElementLocation), that are defined in the Build assembly and not known to Framework.